portfolio/src/main/webapp/script.js

```diff
@@ -57,15 +57,73 @@ function getData() {
     printCom.innerHTML = '';
     comments.forEach((comment) => {
      printCom.appendChild(
-        createCommentFormat("Name: " + comment.username));
-      printCom.appendChild(
-        createCommentFormat("Comment: " + comment.comment));
-    })
+        createElementFormat(comment.username + ":    " + comment.comment, 'li'));
+    });
   });
 }
 
-function createCommentFormat(text) {
-  const liElement = document.createElement('li');
-  liElement.innerText = text;
-  return liElement;
+/**
+ * create element in doc w text and type input
+ */
+function createElementFormat(text, elementType) {
+  const element = document.createElement(elementType);
+  element.innerText = text;
+  return element;
+}
+
+/**
+ * Display the user's login information
+ */
+function showLoginInfo() {
+  fetch('/login').then(response => response.json()).then((user) => {
+    const userInfo = document.getElementById('login-container');
+    userInfo.innerText = '';
+	
+    // display email and logout if logged in, otherwise login
+    if (user.email != null) {
+      userInfo.append(createElementFormat(user.email, 'p'));
+
+      const printLogout = document.createElement('a');
+      printLogout.href = user.url;
+      printLogout.innerText = 'Logout Here';
+
+      userInfo.append(printLogout);
+    } else {
+      const printLogin = document.createElement('a');
+      printLogin.href = user.url;
+      printLogin.innerText = 'Login Here';
+
+      userInfo.append(printLogin);
+    }
+  });
+}
+
+/**
+ * Function that finds if user is logged in, to determine showing comments
+ */
+function commentsOnLoad() {
+  fetch('/login').then(response => response.json()).then((user) => {
+    if (user.email != null) {
+    document.getElementById('comment').style.display = 'block';
+    } else {
+      document.getElementById('comment').style.display = 'none';
+    }
+  });
+}
+
+/**
+ * helper function to load all necessary information "onload"
+ */
+function onLoad() {
+  showLoginInfo();
+  commentsOnLoad();
+  placeEmail()
```

Comment:
So I like the way you've broken down this file into these three different tasks. Eg: show the login link, show the email, and maybe load comments.

But each of these methods fetches the same information from the server. Fetching can be "slow" since it requires waiting for the server to respond, and the server will have to do work for each request even if the requests are identical. So, it's usually recommended to avoid sending repeated fetch requests if you can avoid it.

That does make it harder to have a good decomposition of tasks, as you do here. I'm curious to hear your thoughts on how you'd structure this class so that it both 1) only fetches /login once during onLoad, and 2) still has a clear breakdown of tasks.

Eg: one way to try to have both would be to have one method called in onLoad which fetches from /login, then that method can call each of placeEmail, showLoginInfo, and commentsOnLoad and pass the 'user' object as a parameter.

Reply:
I think I will take your last suggestion. I think I definitely need separate function for the login link but I'm not sure if i need separate function for the comment and email display. Could be clearer in terms of modularity...

Reply:
made one function for display email and comment section but wrote the display email is a helper in there.